### PR TITLE
Dont use default settings for yauzl

### PR DESCRIPTION
**Pull request by rozap**
* Apparently the defaults aren't recommended
* Added a test, from EN-218

Refs EN-218

### CODE REVIEW

File: es6-lib/decoders/kmz.js [hunk 1]
```diff
@@ -33,7 +33,6 @@ class KMZ extends Duplex {
 
     this.on('finish', this._onFinished.bind(this));
     this._zBuffer.on('finish', this._onBuffered.bind(this));
-    this._kmlDecoder = new KML();
```
Comment (kaidamasaki) [inline es6-lib/decoders/kmz.js]: Why was this causing problems?  Were there side effects left over?

File: es6-lib/decoders/kmz.js [hunk 2]
```diff
@@ -78,31 +75,42 @@ class KMZ extends Duplex {
...
       });
   }
 
+  _onError(err) {
+    this.emit('error', err);
+  }
+
   _startPushing() {
     this._isPushing = true;
     var hasOpened = false;
```
Comment (kaidamasaki) [inline es6-lib/decoders/kmz.js]: Is this unused now?

File: es6-lib/services/spatial.js [hunk 3]
```diff
@@ -197,9 +197,10 @@ class SpatialService {
             core.bufferResponse((bufferedResponse) => {
               req.log.info(`Finished upsert response ${upsertResponse.statusCode}`);
               if (bufferedResponse.statusCode > 300) {
+                req.log.error(`Core returned upsert error: ${bufferedResponse}`);
                 return cb({
                   statusCode: 503,
-                  reason: bufferedResponse.body,
+                  reason: "Upstream Error",
```
Comment (kaidamasaki) [inline es6-lib/services/spatial.js]: Is there a reason the response body was dropped?  Could that sometimes fail?

File: es6-lib/services/spatial.js [hunk 4]
```diff
@@ -213,10 +214,11 @@ class SpatialService {
             //
             //It's important to munge this error so core doesn't
             //get blamed for a local exception
+            req.log.error(err.toString());
             upsertRequest.abort();
             return cb({
               statusCode: 500,
-              body: err.toString(),
+              reason: "Internal Error",
```
Comment (kaidamasaki) [inline es6-lib/services/spatial.js]: Again, is there a reason that the error was left off here?

**Comment by kaidamasaki** [comment]
LGTM, though I can't say I followed all of the KML logic.

File: es6-lib/decoders/kmz.js (hunk 1 above)
Comment (rozap) [inline es6-lib/decoders/kmz.js]: yea KML is a stream, which is very side effecty

File: es6-lib/decoders/kmz.js (hunk 2 above)
Comment (rozap) [inline es6-lib/decoders/kmz.js]: no it's used, i added it to remove the duplicated `(err) => this.emit('error', err)` stuff littering the streams

File: es6-lib/services/spatial.js (hunk 3 above)
Comment (rozap) [inline es6-lib/services/spatial.js]: They are JSON errors, but there's no consistency. That coupled with the fact that frontend doesn't handle JSON errors, means the only thing to do at this point is to return a generic error for internal failures and log the actual content. :(

**Pull request merged**